Rubberduck.CodeAnalysis/Inspections/Concrete/UnreachableCaseInspection/ParseTreeValueVisitor.cs

```diff
@@ -286,6 +300,25 @@ private void GetContextValue(ParserRuleContext context, out string declaredTypeN
                 expressionValue = rangeClauseIdentifierReference.IdentifierName;
                 declaredTypeName = GetBaseTypeForDeclaration(declaration);
 
+                if (declaration is ValuedDeclaration valuedDeclaration)
+                {
+                    expressionValue = valuedDeclaration.Expression;
+                    declaredTypeName = GetBaseTypeForDeclaration(declaration);
+                    if (IsVBStringConstant(expressionValue))
```

Comment:
I might be missing something but what if I have a user defined string constant? Shouldn't it get the same treatment as the built-in string constants?

Reply:
I'm intentionally short circuiting the built-in string constants because I've yet to find a workable approach.  My goal with this PR is to address the NREs with respect to non-string built-in constants.  The basic problem to solve for strings is that the built-in  string constants like vbCr have content but are zero-length.  The 'zero-length-ness' is not handled properly by the existing string constant code that addresses user-defined constants.  Handling/Comparing the built-in string constants is (based on my initial attempts) a much more difficult change.